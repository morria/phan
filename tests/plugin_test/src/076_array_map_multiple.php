<?php

/**
 * @param array<string,string> $a
 * @param array<string,stdClass> $b
 * @return array<string,int>
 */
function array_map_multiple_test(array $a, array $b) {
    // Should warn about wrong params and return type
    return array_map(function(stdClass $x, string $y) : array {
        return [$x, $y];
    }, $a, $b);
}
array_map_multiple_test(['x' => 'y'], ['x' => new stdClass()]);
