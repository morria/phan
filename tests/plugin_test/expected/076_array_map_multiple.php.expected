src/076_array_map_multiple.php:10 PhanTypeMismatchReturn Returning type array<int,array> but array_map_multiple_test() is declared to return array<string,int>
src/076_array_map_multiple.php:12 PhanTypeMismatchArgument Argument 1 (x) is string but Closure(\stdClass $x, string $y) : array takes \stdClass defined at src/076_array_map_multiple.php:10
src/076_array_map_multiple.php:12 PhanTypeMismatchArgument Argument 2 (y) is \stdClass but Closure(\stdClass $x, string $y) : array takes string defined at src/076_array_map_multiple.php:10
